### PR TITLE
Add namelist control of PIO options

### LINKED ISSUE #73: Update to Consortium/main; add namelist control of PIO options
Update from Consortium/main to include the following PRs:

- add pio and netcdf error checks https://github.com/CICE-Consortium/CICE/pull/927
- add namelist control of pio optiosn https://github.com/CICE-Consortium/CICE/pull/928

Also includes the following minor PRs
- Fix single channel debug failure, Update github actions testing (#922)
- First round of housekeeping on ice_grid (#921)
- scamn bugfix for nuopc driver (#926)
- Add vorticity as a diagnostic output (#924)
- Fix for ice_mesh_mod with grid variables removed (#929)
- Update PULL_REQUEST_TEMPLATE to request detailed information (#931)
- ug_testing.rst: also mention checking the base suite results (#934)

### PR DESCRIPTION
**Pull request by DeniseWorthen**
Updates to latest Consortium/main. 

- resolves #73

### CODE REVIEW

File: cicecore/cicedyn/infrastructure/ice_grid.F90 [hunk 1]
```diff
@@ -1353,20 +1265,14 @@ subroutine latlongrid
             dyN   (i,j,iblk) = 1.e36_dbl_kind
             dxE   (i,j,iblk) = 1.e36_dbl_kind
             dyE   (i,j,iblk) = 1.e36_dbl_kind
```
Comment (NickSzapiro-NOAA) [inline cicecore/cicedyn/infrastructure/ice_grid.F90:1267]: = bigdbl?

File: cicecore/cicedyn/infrastructure/ice_grid.F90 [hunk 2]
```diff
@@ -4577,7 +4485,8 @@ subroutine get_bathymetry
             do j = 1, ny_block
             do i = 1, nx_block
                k = min(nint(kmt(i,j,iblk)),nlevel)
-               if (k > nlevel) call abort_ice(subname//' kmt gt nlevel error')
+               if (k > nlevel) call abort_ice(subname//' kmt gt nlevel error', &
+                                    file=__FILE__, line=__LINE__)
```
Comment (NickSzapiro-NOAA) [inline cicecore/cicedyn/infrastructure/ice_grid.F90:4489]: this check is not needed. k=min(?,nlevel) already

File: configuration/scripts/machines/Macros.conda_macos [hunk 3]
```diff
@@ -14,7 +14,8 @@ FFLAGS     := -fconvert=big-endian -fbacktrace -ffree-line-length-none
 
 # Additional flags for the Fortran compiler when compiling in debug mode
 ifeq ($(ICE_BLDDEBUG), true)
-  FFLAGS   += -O0 -g -fcheck=bounds -finit-real=nan -fimplicit-none -ffpe-trap=invalid,zero,overflow
+#  FFLAGS   += -O0 -g -fcheck=bounds -finit-real=nan -fimplicit-none -ffpe-trap=invalid,zero,overflow
+  FFLAGS   += -O0 -g -fcheck=bounds -finit-real=nan -fimplicit-none -ffpe-trap=zero,overflow
```
Comment (NickSzapiro-NOAA) [inline configuration/scripts/machines/Macros.conda_macos:18]: why drop invalid?

**Review by NickSzapiro-NOAA — APPROVED** [review]

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by FernandoAndrade-NOAA** [comment]
Testing for #[2145](https://github.com/ufs-community/ufs-weather-model/pull/2145) is complete, please continue with the merge process, thank you

**Comment by FernandoAndrade-NOAA** [comment]
Please also verify review questions should be marked as resolved / answered.

**Pull request merged**